### PR TITLE
[Kotlin] Upgrade jackson version used in the Kotlin client

### LINKED ISSUE #8688: [Kotlin] Jackson is outdated
##### Description

The project is generated with an outdated version of Jackson when language is "kotlin" and serializationLibrary is "jackson". It also references the obsolete JDK8 module for Jackson which is no longer needed  

##### openapi-generator version

5.1.0-SNAPSHOT

##### Command line used for generation

maven plugin configuration:
```xml
<configuration>
    <inputSpec>${project.build.directory}/openapi.json</inputSpec>
    <generatorName>kotlin</generatorName>
    <configOptions>
        <dateLibrary>java8</dateLibrary>
        <java11>true</java11>
        <useCoroutines>true</useCoroutines>
        <collectionType>list</collectionType>
        <enumPropertyNaming>UPPERCASE</enumPropertyNaming>
        <serializationLibrary>jackson</serializationLibrary>
    </configOptions>
    <packageName>org.example</packageName>
    <library>jvm-okhttp4</library>
    <generateApiTests>false</generateApiTests>
    <generateModelTests>false</generateModelTests>
    <generateSupportingFiles>true</generateSupportingFiles>
    <addCompileSourceRoot>true</addCompileSourceRoot>
</configuration>
```
##### Related issues/PRs
#8689 
##### Suggest a fix/enhancement

Adjust mustache templates to use an updated version of Jackson and remove all references to the JDK8 module as all functionality has been merged into jackson-core in the latest versions
I will provide a PR that will solve this issue

### PR DESCRIPTION
**Pull request by zapodot**
To fix #8688 

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
Suggested reviewers: @jimschubert  @dr4ke616  @karismann @Zomzog @andrewemery @4brunu @yutaka0m

### CODE REVIEW

**Comment by 4brunu** [comment]
I'm not a user of `Jackson`, so it's hard for me to review this PR.
I have one question, does `Jackson` require kotlin reflect dependency?

**Comment by zapodot** [comment]
> I'm not a user of `Jackson`, so it's hard for me to review this PR.
> I have one question, does `Jackson` require kotlin reflect dependency?

Yes, according to the pom of jackson-module-kotlin it does depend on kotlin-reflect: https://github.com/FasterXML/jackson-module-kotlin/blob/master/pom.xml

File: modules/openapi-generator/src/main/resources/kotlin-client/jvm-common/infrastructure/Serializer.kt.mustache [hunk 1]
```diff
@@ -22,12 +22,10 @@ import org.threeten.bp.OffsetDateTime
...
-import com.fasterxml.jackson.databind.ObjectMapper
-import com.fasterxml.jackson.databind.SerializationFeature
-import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
-import com.fasterxml.jackson.annotation.JsonInclude
-import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+    import com.fasterxml.jackson.databind.ObjectMapper
+    import com.fasterxml.jackson.databind.SerializationFeature
+    import com.fasterxml.jackson.annotation.JsonInclude
+    import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/kotlin-client/jvm-common/infrastructure/Serializer.kt.mustache:28]: @zapodot thanks for the PR. What about removing these additional spaces?

Comment (zapodot) [reply modules/openapi-generator/src/main/resources/kotlin-client/jvm-common/infrastructure/Serializer.kt.mustache]: Done!

**Review by wing328 — APPROVED** [review]
LGTM but I've not tested it locally

**Comment by wing328** [comment]
@zapodot thanks for the PR, which has been included in the v5.1.0 release: https://twitter.com/oas_generator/status/1373636654024380423

**Pull request merged**